examples/ensemble/plot_forest_hist_grad_boosting_comparison.py

```diff
@@ -0,0 +1,210 @@
+)
+
+for idx, result in enumerate(results):
+    cv_results = result["cv_results"].round(3)
+    model_name = result["model"]
+    param_name = list(param_grids[model_name].keys())[0]
+    cv_results[param_name] = cv_results["param_" + param_name]
+    cv_results["model"] = model_name
+
+    scatter_fig = px.scatter(
+        cv_results,
+        x="mean_fit_time",
+        y="mean_test_score",
+        error_x="std_fit_time",
+        error_y="std_test_score",
+        hover_data=param_name,
+        color="model",
+    )
+    line_fig = px.line(
+        cv_results,
+        x="mean_fit_time",
+        y="mean_test_score",
+    )
+
+    scatter_trace = scatter_fig["data"][0]
+    line_trace = line_fig["data"][0]
+    scatter_trace.update(marker=dict(color=colors_list[idx]))
+    line_trace.update(line=dict(color=colors_list[idx]))
+    fig.add_trace(scatter_trace, row=1, col=1)
+    fig.add_trace(line_trace, row=1, col=1)
+
+    scatter_fig = px.scatter(
+        cv_results,
+        x="mean_score_time",
+        y="mean_test_score",
+        error_x="std_score_time",
+        error_y="std_test_score",
+        hover_data=param_name,
+    )
+    line_fig = px.line(
+        cv_results,
+        x="mean_score_time",
+        y="mean_test_score",
+    )
+
+    scatter_trace = scatter_fig["data"][0]
+    line_trace = line_fig["data"][0]
+    scatter_trace.update(marker=dict(color=colors_list[idx]))
+    line_trace.update(line=dict(color=colors_list[idx]))
+    fig.add_trace(scatter_trace, row=1, col=2)
+    fig.add_trace(line_trace, row=1, col=2)
+
+fig.update_layout(
+    xaxis=dict(title="Train time (s) - lower is better"),
+    yaxis=dict(title="Test R2 score - higher is better"),
+    xaxis2=dict(title="Predict time (s) - lower is better"),
+    legend=dict(x=0.72, y=0.05, traceorder="normal", borderwidth=1),
+    title=dict(x=0.5, text="Speed-score trade-off of tree-based ensembles"),
+)
+
+# %%
+# Both HGBT and RF models improve when increasing the number of trees in the
+# ensemble. However, the scores reach a plateau where adding new trees just
+# makes fitting and scoring slower.
+#
+# Unlike RF, HGBT models offer an early-stopping option (See
+# :ref:`sphx_glr_auto_examples_ensemble_plot_gradient_boosting_early_stopping.py`)
+# to avoid adding new unnecessary trees. Internally, the algorithm uses an
+# out-of-sample set to compute the generalization performance of the model at
+# each addition of a tree. Thus, if the generalization performance is not
+# improving for more than `n_iter_no_change` iterations, it stops adding trees.
+#
+# In this example HGBT performs uniformly better than RF in terms of test score
+# and computing time (train and predict). Nevertheless, the training and
+# predicting time of RF can be reduced using the `n_jobs` parameter, as
+# mentioned above. Overall, the performance of HGBT versus parallelized RF
+# depends on the specific characteristics of the dataset and the modeling task.
```

Comment:
Here is a suggestion to attempt to make this paragraph more robust to the outcome of the run on various machines:

```suggestion
# Note that the results shown on the above plot can change sightly across
# runs and even more significantly when running on other machines: try to
# run this example on your own local machine.
#
# Overall, one should often observe that the Histogram-based gradient boosting
# model to uniformly dominate the Random Forest model in the "test score vs
# training speed trade-off" (the HGBDT curve should be on the top left of the
# RF curve, without ever crossing). The "test score vs prediction speed"
# trade-off can also be more disputed but it's most often favorable to HGBDT.
```